### PR TITLE
fix(js): .swcrc path option should follow existing conventions

### PR DESCRIPTION
**Pull request by djgrant**
## Current Behavior

While testing #10118 locally, I noticed a couple of potential issues that should perhaps be addressed before release:
- `swcrcPath` is not named consistently with other options e.g. `tsConfig`, `packageJson`
- `swcrcPath` is resolved relative to the project roots, whereas `tsConfig` and `packageJson` are resolved relative to the workspace root

## Expected Behavior

~The implementation follows the same approach as the `main` option – the option is resolved relative to the project path if the project path is not in the string.~

Update: `swcrc` path is alway resolved relative to the workspace root. 

```diff 
{
  "targets": {
    "build": {
      "executor": "@nrwl/js:swc",
      "options": {
         // path relative to project root",
-        "swcrcPath": "config/swcrc.json"
         // path relative to workspace root (consistent with `tsConfig` and `packageJson`
+        "swcrc": "/packages/my-package/config/swcrc.json",
      }
    }
  }
}
````

The reason for following this implentation is that the default value defined in `schema.json` is static so cannot include the relative project path (unless we introduce some kind of string replacement e.g. `$projectRoot/.lib.swcrc`).  This is not an issue with`tsConfig` and `packageJson` as they are required options. 

## Related Issue(s)

#10058

### CODE REVIEW

**Comment by djgrant** [comment]
@AgentEnder noticed a couple of issues with #10118. I don't think this is the right solution because the behaviour is unpredictable (e.g. it looks like you can provide a path relative to the workspace root, but it always resolves ot the package root) but wanted to put something out there to get the ball rolling. 

If having a default value in `schema.json` is a fixed constraint could we do one of the following:
- support string substitution e.g. `"default": "$projectRoot/.lib.swcrc"`
- support a default flag e.g. `"default": true`, and define the default value in swc.impl.ts

I appreciate there are different types of misdirection from each of these solutions.

**Comment by djgrant** [comment]
@AgentEnder thanks for the guidance on this. I've had a crack at it and tested locally. Let me know if there's anything I've missed.

File: packages/js/migrations.json [hunk 1]
```diff
@@ -23,6 +23,12 @@
       "version": "14.0.0-beta.2",
       "description": "Exclude jest config from .lib.swcrc",
       "factory": "./src/migrations/update-14-0-0/exclude-jest-config-swcrc"
+    },
+    "update-swcrc-path": {
+      "cli": "nx",
+      "version": "14.1.1-beta.0",
```
Comment (AgentEnder) [inline packages/js/migrations.json:29]: 14.1.1 has been released already, can you bump this to 14.1.5-beta.0?

**Review by AgentEnder — CHANGES_REQUESTED** [review]
This looks good, do you care to update the migration version and we can get it merged?

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**